airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -103,7 +112,10 @@ def __init__(
         self.handler: Union[logging.FileHandler, logging.StreamHandler]  # type: ignore[assignment]
 
     def _render_log_id(self, ti: TaskInstance, try_number: int) -> str:
-        dag_run = ti.get_dagrun()
+        with create_session() as session:
+            dag_run = ti.get_dagrun(session=session)
+            log_id_template = dag_run.get_log_template(session=session).elasticsearch_id
```

Comment:
This needs more work to make it backwards compatible, or we bump the min Airflow version.

Reply:
Ah, right. I think we can simply detect whether there’s a `LogTemplate` model.

Reply:
Added a compat layer.